File: 01/2/main.py
import itertools

f = open('input')

lst_iter = itertools.cycle(
    map(lambda x: int(x[:-1]), f)
)

temp = 0
freq = set()

for i in lst_iter:
    if temp in freq:
        print(temp)
        break
    freq.add(temp)
    temp += i

f.close()

File: 01/2/main.rs
use std::fs::File;
use std::io::{BufRead, BufReader};
use std::collections::HashSet;

type Result<T> = ::std::result::Result<T, Box<::std::error::Error>>;

fn main() -> Result<()> {
    let file = File::open("input")?;
    let f = BufReader::new(file);

    let mut ints = Vec::new();

    for line in f.lines() {
        let i = line?.parse::<i32>()?;
        ints.push(i);
    }


    let mut freq = HashSet::new();
    let mut temp = 0;

    for i in ints.iter().cycle() {
        if freq.contains(&temp) {
            println!("{}", temp);
            break
        }
        freq.insert(temp);
        temp += i;
    }
    Ok(())
}
